config-controller/pkg/client/client_test.go

```diff
@@ -128,6 +128,48 @@ func TestCachedClientGet(t *testing.T) {
 	assert.False(t, exists, "Policy exists when it should not")
 }
 
+// TestCachedClientDelete validates that the cached client deletes policies as expected
+func TestCachedClientDelete(t *testing.T) {
+	newPolicy := storage.Policy{
+		Name:            "This is a new policy",
+		Description:     "A really good description",
+		LifecycleStages: []storage.LifecycleStage{storage.LifecycleStage_BUILD},
+		Severity:        storage.Severity_CRITICAL_SEVERITY,
+		PolicySections: []*storage.PolicySection{{
+			SectionName: "Section A",
+			PolicyGroups: []*storage.PolicyGroup{{
+				FieldName: "Image",
+				Values: []*storage.PolicyValue{{
+					Value: "hello",
+				}},
+			}},
+		}},
+	}
+
+	mockPolicyToReturn := newPolicy.CloneVT()
+	mockPolicyToReturn.Id = "abc123"
+	mockPolicyToReturn.Source = storage.PolicySource_DECLARATIVE
```

Comment:
would it be valuable to try deleting a policy with an imperative source and assert that there was an error?

Reply:
I actually ended up testing that before I added the line to mark declarative :-), yes will add it as a test